### ISSUE TITLE
[Bug] Build fails in Astro v5.x: `Cannot read properties of undefined (reading 'name')`

### ISSUE THREAD

**Issue by mxmason** [vanilla-extract-css/vanilla-extract#1509]
### Describe the bug

As of Astro 5.0, this plugin breaks Vite builds. Non-Astro projects using Vite v6 build just fine, and downgrading to Astro@4 also works just fine. To reproduce:

1. Clone the attached repo
2. run `npm ci`
3. run `npm run build`

### Reproduction

https://github.com/mxmason/astro-vanilla-extract-bug

### System Info

```shell
System:
    OS: macOS 15.1.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 95.75 MB / 32.00 GB
    Shell: 5.9 - /opt/homebrew/opt/zsh/bin/zsh
  Binaries:
    Node: 20.14.0 - ~/.fnm/node-versions/v20.14.0/installation/bin/node
    npm: 10.9.0 - ~/.fnm/node-versions/v20.14.0/installation/bin/npm
  Browsers:
    Chrome: 131.0.6778.87
    Safari: 18.1.1
    Safari Technology Preview: 18.2
  npmPackages:
    @vanilla-extract/css: ^1.16.1 => 1.16.1 
    @vanilla-extract/vite-plugin: ^4.0.18 => 4.0.18 
    astro: ^5.0.2 => 5.0.2
```

### Used Package Manager

npm

### Logs

```shell
Cannot read properties of undefined (reading 'name')
file: /user/dev/astro-vanilla-extract-bug/src/test.css.ts
  Stack trace:
    at TransformPluginContext.transform (file:///user/dev/astro-vanilla-extract-bug/node_modules/astro/dist/vite-plugin-astro/index.js:162:30)
    at loadAndTransform (file:///user/dev/astro-vanilla-extract-bug/node_modules/vite/dist/node/chunks/dep-CB_7IfJ-.js:51929:49)
    at async ViteNodeServer._fetchModule (file:///user/dev/astro-vanilla-extract-bug/node_modules/vite-node/dist/server.mjs:391:17)
    at async ViteNodeRunnerWithContext.cachedRequest (file:///user/dev/astro-vanilla-extract-bug/node_modules/vite-node/dist/client.mjs:189:14)
    at async /user/dev/astro-vanilla-extract-bug/node_modules/@vanilla-extract/integration/dist/vanilla-extract-integration.cjs.prod.js:807:29
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by LekoArts** [comment]
I reported this with Astro at https://github.com/withastro/astro/issues/12614 and they were able to unblock me + give pointers of what needs changing.

What worked for me to get it working:

```json
"pnpm": {
  "overrides": {
    "vite": "^6.0.2"
  }
}
```

What needs to be changed properly (non exhaustive list):

- Update https://github.com/vanilla-extract-css/vanilla-extract/blob/dde9bf2d15120e546be372390fa562384a48f99b/packages/integration/package.json#L28-L29
- Update https://github.com/vanilla-extract-css/vanilla-extract/blob/master/packages/vite-plugin/src/index.ts#L144

**Comment by danielberndt** [comment]
Can confirm that adding

```
  "overrides": {
    "@vanilla-extract/integration": {
      "vite": "^6.0.3"
    }
  },
```

to my `package.json` in a `npm install` based workflow got it working again.

**Comment by jojojojojoj5564656465465** [comment]
does vanilla extract will fix it or i need to change my config ? i can stay in astro 4.16 otherwise

**Comment by askoufis** [comment]
> does vanilla extract will fix it or i need to change my config ? i can stay in astro 4.16 otherwise

Yes Vanilla Extract will fix this. We'll support vite 6 and that should address this issue. However, I'm going to wait until `vite-node` v3 comes out alongside `vitest` v3 (currently still in beta) before updating, because [`vite-node` v3 will add support for `vite` v6](https://github.com/vitest-dev/vitest/blame/4e60333dc7235704f96314c34ca510e3901fe61f/packages/vite-node/package.json#L89).

**Comment by danielberndt** [comment]
Another heads up: there seem to be compatibility issues with `@astrojs/mdx v4`

I've got this minimal `astro.config.mjs`:

```js
export default defineConfig({
  vite: {
    plugins: [vanillaExtractPlugin()],
  },
  integrations: [mdx()],
});
```

leading to a really weird error message:

![image](https://github.com/user-attachments/assets/0100f523-6c45-4e62-a397-9a02df335cfb)


Does it try to parse the `.mdx` file as plain javascript?

Here's the full stack trace

```
TypeError: (0 , __vite_ssr_import_0__.createAstro) is not a function
    at /[PROJECT_DIR]/node_modules/astro/components/Image.astro:1:1
    at async ESModulesEvaluator.runInlinedModule (file:///[PROJECT_DIR]/node_modules/vite/dist/node/module-runner.js:1053:5)
    at async SSRCompatModuleRunner.directRequest (file:///[PROJECT_DIR]/node_modules/vite/dist/node/module-runner.js:1279:61)
    at async SSRCompatModuleRunner.directRequest (file:///[PROJECT_DIR]/node_modules/vite/dist/node/chunks/dep-yUJfKD1i.js:30612:23)
    at async SSRCompatModuleRunner.cachedRequest (file:///[PROJECT_DIR]/node_modules/vite/dist/node/module-runner.js:1174:76)
    at async eval (/[PROJECT_DIR]/astro:assets:4:6)
    at async ESModulesEvaluator.runInlinedModule (file:///[PROJECT_DIR]/node_modules/vite/dist/node/module-runner.js:1053:5)
    at async SSRCompatModuleRunner.directRequest (file:///[PROJECT_DIR]/node_modules/vite/dist/node/module-runner.js:1279:61)
    at async SSRCompatModuleRunner.directRequest (file:///[PROJECT_DIR]/node_modules/vite/dist/node/chunks/dep-yUJfKD1i.js:30612:23)
    at async SSRCompatModuleRunner.cachedRequest (file:///[PROJECT_DIR]/node_modules/vite/dist/node/module-runner.js:1174:76)
```

versions used:
```
@astrojs/mdx: 4.0.2
@vanilla-extract/vite-plugin: 4.0.18
astro: 5.0.5
```

and 
```
"overrides": {
    "@vanilla-extract/integration": {
      "vite": "^6.0.3"
    },
    "@vanilla-extract/vite-plugin": {
      "vite": "^6.0.3"
    }
  },
```

**Comment by sbuys** [comment]
@danielberndt any chance you've found a workout for this?

**Comment by jojojojojoj5564656465465** [comment]
You need to wait vanilla extract update > @danielberndt any chance you've found a workaround for this?

**Comment by askoufis** [comment]
The Astro + MDX issue may be unrelated to the vite v6 upgrade. I'd encourage making a separate issue that includes a reproduction with vite v5 if the issue still persists on that version.

**Comment by ancashoria** [comment]
> > does vanilla extract will fix it or i need to change my config ? i can stay in astro 4.16 otherwise
> 
> Yes Vanilla Extract will fix this. We'll support vite 6 and that should address this issue. However, I'm going to wait until `vite-node` v3 comes out alongside `vitest` v3 (currently still in beta) before updating, because [`vite-node` v3 will add support for `vite` v6](https://github.com/vitest-dev/vitest/blame/4e60333dc7235704f96314c34ca510e3901fe61f/packages/vite-node/package.json#L89) (v2 only support vite v5).

`vite-node` v3 and Vitest v3 have launched! Horray
